### PR TITLE
Crumbling Armor addition and Silent Girl Tweak

### PR DESCRIPTION
**Pull request by dsmith328**
## About The Pull Request
Adds Crumbling Armor and it's EGO!

## Why It's Good For The Game
Teth with PALE DAMAGE WEAPON! Let's go!!! Also, the rudimentary EGO gift that gives +20 Justice for -20 Fortitude is cool.

## Changelog
:cl:
add: Crumbling Armored
add: Life for a Daredevil
/:cl:

### CODE REVIEW

**Comment by dsmith328** [comment]
Accidentally merged this with the Silent Girl Tweak, so I'm pulling that one and changing this one.

**Comment by dsmith328** [comment]
Daredevil
Crumbling
crumblingdesc
STATSBABY

**Comment by Kitsunemitsu** [comment]
Can we get a records page for the Armor?

**Comment by dsmith328** [comment]
> Can we get a records page for the Armor?

Done!

**Comment by EgorDinamit** [comment]
The overlay should be on the layer below mob, so it doesn't look that weird. Sprite itself is rather weird, too.

File: code/modules/mob/living/simple_animal/abnormality/teth/crumbling_armor.dm [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	maxHealth = 600
+	health = 600
+	threat_level = TETH_LEVEL
+	work_chances = list(
+		ABNORMALITY_WORK_INSTINCT = list(50, 50, 55, 55, 60),
+		ABNORMALITY_WORK_INSIGHT = 40,
+		ABNORMALITY_WORK_ATTACHMENT = 0,
+		ABNORMALITY_WORK_REPRESSION = list(60, 60, 65, 65, 70)
+			)
+	pixel_y = -8
```
Comment (EgorDinamit) [inline code/modules/mob/living/simple_animal/abnormality/teth/crumbling_armor.dm:15]: No need to offset it by y, it makes it looks weird.

File: code/modules/mob/living/simple_animal/abnormality/teth/crumbling_armor.dm [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	var/buffIcon = 'ModularTegustation/Teguicons/32x48.dmi'
+	var/mob/living/carbon/human/courageous = list()
+	var/mob/living/carbon/human/reckless = list()
+	var/mob/living/carbon/human/foolish = list()
+	var/mob/living/carbon/human/suicidal = list()
+
+/mob/living/simple_animal/hostile/abnormality/crumbling_armor/Initialize(mapload)
+	. = ..()
+	// Megalovania?
+	if (prob(1))
```
Comment (EgorDinamit) [inline code/modules/mob/living/simple_animal/abnormality/teth/crumbling_armor.dm:34]: Another Kirie idea..?

File: code/modules/mob/living/simple_animal/abnormality/teth/crumbling_armor.dm [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+		icon_state = "megalovania"
+
+/mob/living/simple_animal/hostile/abnormality/crumbling_armor/proc/cut_head(datum/source, datum/abnormality/datum_sent, mob/living/carbon/human/user, work_type)
```
Comment (EgorDinamit) [inline code/modules/mob/living/simple_animal/abnormality/teth/crumbling_armor.dm:37]: You forgot SIGNAL_HANDLER at the beginning of the proc.

File: code/modules/mob/living/simple_animal/abnormality/teth/crumbling_armor.dm [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+	ego_list = list(
+		/datum/ego_datum/weapon/daredevil,
+		/datum/ego_datum/armor/daredevil
+		)
+
+	var/buffIcon = 'ModularTegustation/Teguicons/32x48.dmi'
+	var/mob/living/carbon/human/courageous = list()
+	var/mob/living/carbon/human/reckless = list()
+	var/mob/living/carbon/human/foolish = list()
+	var/mob/living/carbon/human/suicidal = list()
```
Comment (EgorDinamit) [inline code/modules/mob/living/simple_animal/abnormality/teth/crumbling_armor.dm:29]: Looks like a waste of space.
Make it an associative list instead.

**Comment by EgorDinamit** [comment]
It also seems that the inhand sword and worn icons for armor are offset, fix it.

File: code/modules/mob/living/simple_animal/abnormality/teth/crumbling_armor.dm (hunk 4 above)
Comment (dsmith328) [reply code/modules/mob/living/simple_animal/abnormality/teth/crumbling_armor.dm:29]: > The overlay should be on the layer below mob, so it doesn't look that weird. Sprite itself is rather weird, too.

I wanted a better sprite for it but I was told "it was fine". I'll request one in Spriter General

**Comment by dsmith328** [comment]
> The overlay should be on the layer below mob, so it doesn't look that weird. Sprite itself is rather weird, too.

image
Images were old, this was done a while ago

**Comment by dsmith328** [comment]
> It also seems that the inhand sword and worn icons for armor are offset, fix it.

image
Fixed

**Pull request merged**